### ISSUE TITLE
Limit by maximum distance

### ISSUE THREAD

**Issue by brean** [brean/python-pathfinding#20]
As suggested [here](https://github.com/qiao/PathFinding.js/issues/117#issuecomment-481342450), we could also limit the runtime by a maximum path length instead of just the number of iterations.

**Comment by UP929312** [comment]
So instead of using max_runs (which currently defaults to infinity), we could get a max_depth parameter, for one vector, rather than a general spread? I'd certainly use this feature

**Comment by brean** [comment]
The idea was simple: Use the distance from the start to the next nodes (what we are calculating anyway) and if its above a certain threshold do not go into the neighbors of that node (we can call it max_depth). Should be a 1-liner...
But we could also have a parameter to limit the angle in which we search.

**Comment by UP929312** [comment]
> The idea was simple: Use the distance from the start to the next nodes (what we are calculating anyway) and if its above a certain threshold do not go into the neighbors of that node (we can call it max_depth). Should be a 1-liner... But we could also have a parameter to limit the angle in which we search.

Sounds good to me, not sure what the angle bit means, but max_depth being a simple addition is exciting

**Comment by UP929312** [comment]
Hey @brean, know you're a busy person, with not much time
This feature would be super helpful for larger maps as well, saves having to make lots of calculations.
Has anything been done to progress this? Merely sharing my excitement

**Comment by brean** [comment]
No, I have not worked on this any further, but I have some more thoughts on the topic of handling larger maps:
1. The number of iterations is the better heutristic, as the limiting factor is the iterations on the CPU and because we move along a grid with a fixed distance most of the time the distance roughly translate into the number of iterations. So number of iterations is already defining the max. distance but dependent on the number of iterations (CPU-cycles) you have and not an arbitrary distance.
2. based on your big grid sample a smaller grid (e.g. only a 4th or 8th of the size) and find the shortest path on that map, afterwards run an optimization algorithm on the path to avoid only moving along the roughs grid edges. Note that this might not find the shortest path as you have to mark the whole area as non-passible if there is just one obstacle in your detailed map.
3. for special cases like implementing a motorway/highway or some other logic handling long distances on a map you need to introduce additional knowledge into the algorithms and want probably to write a custom algorithm instead of using this.
4. As discussed in #53 getting faster would require a rewrite of the library with numba or other libraries which might shift the focus from supporting general pathfinding to less generic solutions.
